https://github.com/MisterTea/EternalTerminal/pull/562.patch

From d39008320d401264770636abe9d77ba69e65c620 Mon Sep 17 00:00:00 2001
From: Krul Ceter <kruceter@proton.me>
Date: Sat, 21 Jan 2023 07:36:17 +0300
Subject: [PATCH] Add BUILD_TESTING to make tests optional by choice

The option is enabled by default, so it should not disrupt the default
behavior.
---
 CMakeLists.txt | 48 ++++++++++++++++++++++++++----------------------
 1 file changed, 26 insertions(+), 22 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef239d3e..2368dfc2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,6 +165,7 @@ endif()
 
 option(CODE_COVERAGE "Enable code coverage" OFF)
 option(FUZZING "Enable builds for fuzz testing" OFF)
+option(BUILD_TESTING "Build tests" ON)
 option(DISABLE_CRASH_LOG "Disable installing easylogging crash handler" OFF)
 
 add_definitions(-DET_VERSION="${PROJECT_VERSION}")
@@ -521,30 +522,33 @@ else(WIN32)
       ${CORE_LIBRARIES})
   decorate_target(htmd)
 
-  enable_testing()
+  if(BUILD_TESTING)
+    enable_testing()
 
-  file(GLOB TEST_SRCS test/*Test.cpp)
-  add_executable(
-  et-test
-  ${TEST_SRCS}
-  test/Main.cpp
-  )
-  add_dependencies(et-test TerminalCommon et-lib)
-  target_link_libraries(
+    file(GLOB TEST_SRCS test/*Test.cpp)
+    add_executable(
     et-test
-    TerminalCommon
-    et-lib
-    ${CMAKE_THREAD_LIBS_INIT}
-    ${PROTOBUF_LIBS}
-    ${sodium_LIBRARY_RELEASE}
-    ${SELINUX_LIBRARIES}
-    ${UTEMPTER_LIBRARIES}
-    ${Boost_LIBRARIES}
-      ${CORE_LIBRARIES})
-  add_test(et-test et-test)
-  decorate_target(et-test)
+    ${TEST_SRCS}
+    test/Main.cpp
+    )
 
-  if(FUZZING)
+    add_dependencies(et-test TerminalCommon et-lib)
+    target_link_libraries(
+      et-test
+      TerminalCommon
+      et-lib
+      ${CMAKE_THREAD_LIBS_INIT}
+      ${PROTOBUF_LIBS}
+      ${sodium_LIBRARY_RELEASE}
+      ${SELINUX_LIBRARIES}
+      ${UTEMPTER_LIBRARIES}
+      ${Boost_LIBRARIES}
+        ${CORE_LIBRARIES})
+    add_test(et-test et-test)
+    decorate_target(et-test)
+  endif()
+
+  if(BUILD_TESTING AND FUZZING)
     add_executable(
     TerminalServerFuzzer
     test/TerminalServerFuzzer.cpp
@@ -582,7 +586,7 @@ else(WIN32)
       ${Boost_LIBRARIES}
         ${CORE_LIBRARIES})
     decorate_fuzzer(TerminalServerRouterFuzzer)
-  endif(FUZZING)
+  endif(BUILD_TESTING AND FUZZING)
 
   install(
     TARGETS etserver etterminal et htm htmd
